packages/block-editor/src/components/use-block-sync/index.js

```diff
@@ -174,9 +204,26 @@ export default function useBlockSync( {
 				const updateParent = isPersistent
 					? onChangeRef.current
 					: onInputRef.current;
-				updateParent( blocks, {
-					selectionStart: getSelectionStart(),
-					selectionEnd: getSelectionEnd(),
+
+				const outgoingBlocks = shouldMapBlockIds()
```

Comment:
I've noticed that you call `shouldMapBlockIds()` 3 times between this line and line 220. I think it would be more performance-friendly to just run it once and then reuse the result.

Reply:
7cfc4ac